codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -660,18 +655,30 @@ impl Event {
         human_cfg_event: EventConfig,
         opt_abi: &Option<EvmAbi>,
     ) -> Result<Self> {
-        let event = Event::get_abi_event(&human_cfg_event.event, opt_abi)?.into();
+        let event: NormalizedEthAbiEvent =
+            Event::get_abi_event(&human_cfg_event.event, opt_abi)?.into();
 
         Ok(Event {
+            name: human_cfg_event.name.unwrap_or(event.0.name.to_owned()),
```

Comment:
I'll add validation in another PR together with the fix for another reported bug where codegen fails with contract name containing dashes.